src/sentry/utils/query.py

```diff
@@ -93,14 +93,19 @@ def __iter__(self):
                 cb(results)
 
             for result in results:
-                if result == last_object:
+                if last_object_pk is not None and result.pk == last_object_pk:
```

Comment:
I know this isn't a new line, but this doesn't seem like a very reliable way of handling pagination for non-unique columns (assuming this intended to remove duplicates based on the comment on L75) since the primary key isn't actually used as part of the `ORDER BY` clause so it (and other items already returned) are liable to occur at any position in the result set.

Reply:
Yup, it's not at all. :) I don't feel like addressing that at all here. I also don't think we order by anything other than pk, at least as far as I can tell.

Reply:
Yeah, wasn't suggesting you change it here, just articulating my surprise/confusion when I tried to understand this change